selfservice/hook/web_hook.go

```diff
@@ -307,9 +346,48 @@ func doHttpCall(method string, url string, as AuthStrategy, body io.Reader) erro
 
 	if err != nil {
 		return err
-	} else if resp.StatusCode >= 400 {
+	} else if resp.StatusCode >= http.StatusBadRequest {
+		if interrupt {
+			if err = parseResponse(resp); err != nil {
+				return err
+			}
+		}
 		return fmt.Errorf("web hook failed with status code %v", resp.StatusCode)
 	}
 
 	return nil
 }
+
+func parseResponse(resp *http.Response) error {
+	if resp == nil {
+		return fmt.Errorf("empty response provided from the webhook")
+	}
+
+	body, err := io.ReadAll(resp.Body)
```

Comment:
`io.ReadAll` doesn't exists. You probably meant `ioutil.ReadAll` ?

Reply:
Nope: https://pkg.go.dev/io#ReadAll

Reply:
Ok, they introduced it in 1.16 🙂